llvm/test/Transforms/LoadStoreVectorizer/AArch64/pr37865.ll

```diff
@@ -0,0 +1,14 @@
+; REQUIRES: asserts
+; RUN: not --crash opt -mtriple=aarch64 -passes=load-store-vectorizer -disable-output %s
+
+define i32 @load_cycle(ptr %x) {
+; CHECK: Unexpected cycle while re-ordering instructions
+entry:
+  %gep.x.1 = getelementptr inbounds [2 x i32], ptr %x, i32 0, i32 1
+  %load.x.1 = load i32, ptr %gep.x.1
+  %rem = urem i32 %load.x.1, 1
+  %gep.x.2 = getelementptr inbounds [2 x i32], ptr %x, i32 %rem, i32 0
```

Comment:
I think this just failed to check that one of the vectorization pointer candidates actually depends on the other load, so these shouldn't be vectorization candidates

Reply:
Right, I figured that I have to fix the creation of the Chain.